From 36ab2927bf0ce587ab43190664798c335877e4c1 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <sanyog.r.kale@intel.com>
Date: Wed, 23 Nov 2016 11:18:29 +0530
Subject: [PATCH 4035/4301] SoundWire: Master driver header file changes for
 BRA feature

commit 670740f33f582167d6ea03a9d0d19ed7d4c8408b from
git://git@github.com:01org/linux-intel-4.9.git

This patch includes:
	- Data structure required for BRA operations.
	- BRA ops definition.
	- Defines used by Master driver for BRA operations.

Signed-off-by: Sanyog Kale <sanyog.r.kale@intel.com>
---
 drivers/sdw/sdw_cnl_priv.h  |   32 ++++++++++++++++++++++++++++++++
 include/linux/sdw/sdw_cnl.h |   29 +++++++++++++++++++++++++++++
 2 files changed, 61 insertions(+), 0 deletions(-)

diff --git a/drivers/sdw/sdw_cnl_priv.h b/drivers/sdw/sdw_cnl_priv.h
index 6d5594f..82d0d12 100644
--- a/drivers/sdw/sdw_cnl_priv.h
+++ b/drivers/sdw/sdw_cnl_priv.h
@@ -221,6 +221,8 @@
 #define PDINCONFIG_CHANNEL_MASK_MASK		0xFF
 #define PDINCONFIG_PORT_NUMBER_SHIFT		0x0
 #define PDINCONFIG_PORT_NUMBER_MASK		0x1F
+#define PDINCONFIG_PORT_SOFT_RESET_SHIFT	0x18
+#define PDINCONFIG_PORT_SOFT_RESET		0x1F
 
 #define DPN_CONFIG_WL_SHIFT			0x8
 #define DPN_CONFIG_WL_MASK			0x1F
@@ -342,4 +344,34 @@
 #define CNL_STRMZCFG_CHAN_SHIFT			16
 #define CNL_STRMZCFG_CHAN_MASK			0xF
 
+#define SDW_BRA_HEADER_SIZE_PDI             12 /* In bytes */
+#define SDW_BRA_HEADER_CRC_SIZE_PDI         4 /* In bytes */
+#define SDW_BRA_DATA_CRC_SIZE_PDI           4 /* In bytes */
+#define SDW_BRA_HEADER_RESP_SIZE_PDI        4 /* In bytes */
+#define SDW_BRA_FOOTER_RESP_SIZE_PDI        4 /* In bytes */
+#define SDW_BRA_PADDING_SZ_PDI              4 /* In bytes */
+#define SDW_BRA_HEADER_TOTAL_SZ_PDI         16 /* In bytes */
+
+#define SDW_BRA_SOP_EOP_PDI_STRT_VALUE	0x4
+#define SDW_BRA_SOP_EOP_PDI_END_VALUE	0x2
+#define SDW_BRA_SOP_EOP_PDI_MASK	0x1F
+#define SDW_BRA_SOP_EOP_PDI_SHIFT	5
+
+#define SDW_BRA_STRM_ID_BLK_OUT		3
+#define SDW_BRA_STRM_ID_BLK_IN		4
+
+#define SDW_BRA_PDI_TX_ID		0
+#define SDW_BRA_PDI_RX_ID		1
+
+#define SDW_BRA_SOFT_RESET		0x1
+#define SDW_BRA_BULK_ENABLE		1
+#define SDW_BRA_BLK_EN_MASK		0xFFFEFFFF
+#define SDW_BRA_BLK_EN_SHIFT		16
+
+#define SDW_BRA_ROLLINGID_PDI_INDX	3
+#define SDW_BRA_ROLLINGID_PDI_MASK	0xF
+#define SDW_BRA_ROLLINGID_PDI_SHIFT	0
+
+#define SDW_PCM_STRM_START_INDEX	0x2
+
 #endif /* _LINUX_SDW_CNL_H */
diff --git a/include/linux/sdw/sdw_cnl.h b/include/linux/sdw/sdw_cnl.h
index a2d8ffd..1a40244 100644
--- a/include/linux/sdw/sdw_cnl.h
+++ b/include/linux/sdw/sdw_cnl.h
@@ -73,6 +73,33 @@ struct cnl_sdw_port {
 	struct cnl_sdw_pdi_stream *pdi_stream;
 };
 
+struct bra_packet_info {
+	u8 packet_num;
+	u8 num_data_bytes;
+};
+
+struct bra_info {
+	unsigned int mstr_num;
+	u8 *tx_ptr;
+	u8 *rx_ptr;
+	unsigned int tx_block_size;
+	unsigned int rx_block_size;
+	u8 valid_packets;
+	struct bra_packet_info *packet_info;
+};
+
+struct cnl_bra_operation {
+	int (*bra_platform_setup)(void *context, bool is_enable,
+						struct bra_info *info);
+	int (*bra_platform_xfer)(void *context, bool is_enable,
+						struct bra_info *info);
+};
+
+struct cnl_sdw_bra_cfg {
+	void *drv_data;
+	struct cnl_bra_operation *bra_ops;
+};
+
 struct cnl_sdw_data {
 	/* SoundWire IP registers per instance */
 	void __iomem *sdw_regs;
@@ -84,6 +111,8 @@ struct cnl_sdw_data {
 	int irq;
 	/* Instance id */
 	int inst_id;
+	/* BRA data pointer */
+	struct cnl_sdw_bra_cfg *bra_data;
 };
 
 struct cnl_sdw_port *cnl_sdw_alloc_port(struct sdw_master *mstr, int ch_count,
-- 
1.7.5.4

